packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -541,7 +557,10 @@ export function getValidatorApi({
       if (fullBlock && blindedBlock) {
         switch (builderSelection) {
           case routes.validator.BuilderSelection.MaxProfit: {
-            if (blockValueEngine >= blockValueBuilder) {
+            if (
+              builderBoostFactor === MAX_BUILDER_BOOST_FACTOR ||
```

Comment:
This does not look right, if it is the max value it should pick builder block, not engine

> `builder_comparison_factor=2**64 - 1`: prefer the builder payload unless an error or beacon node health check makes it unviable.

Reply:
oops, thanks for catching

Reply:
updated